### DISCUSSION TITLE
Late sleep problem

### DISCUSSION THREAD

**Discussion by zee-doubleu-ee** [ANLAB-KAIST/KENSv3#219] [Project Q&A 2. Transport Layer (TCP)] [answered]
![Screenshot_20240504_232335](https://github.com/ANLAB-KAIST/KENSv3/assets/73657186/46404984-5bd9-4d61-b2d0-1b3cb3424048)
Unreliable test cases are failing here. Is it necessary to pass this condition? I think the problem here is probably because I have exponential backoff in retransmission (including syn retransmission).

**Comment by Pusnow** [comment, marked as answer]
You must pass all the test cases to get full credit. You don't have to implement congestion control for PA2. Please try to implement flow control first.
